packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -140,8 +144,9 @@ export class PersonalizeMiddleware {
   private handler = async (req: NextRequest, res?: NextResponse): Promise<NextResponse> => {
     const pathname = req.nextUrl.pathname;
     const language = req.nextUrl.locale || req.nextUrl.defaultLocale || 'en';
-    let browserId = this.getBrowserId(req);
+    const pointOfSale = this.config.getPointOfSale(language);
```

Comment:
Don't think there's a huge performance hit here, but we could delay this call until we need it (i.e. just before executeExperience call).